### PR TITLE
GL_SRGB_EXT format is valid texture internal format

### PR DESCRIPTION
**Pull request by xhcao**
Please help to review it, thank you. @zhenyao @qkmiao @Richard-Yunchao

### CODE REVIEW

File: sdk/tests/conformance/extensions/ext-sRGB.html [hunk 1]
```diff
@@ -311,7 +311,7 @@
   var formatString;
   var validFormat;
   switch (format) {
-    case ext.SRGB_EXT: formatString = "sRGB"; validFormat = false; break;
+    case ext.SRGB_EXT: formatString = "sRGB"; validFormat = true; break;
```
Comment (zhenyao) [inline sdk/tests/conformance/extensions/ext-sRGB.html]: SRGB is not color renderable, so framebuffer will not be COMPLETE.  It is correct to set validFormat to false.

File: sdk/tests/conformance/extensions/ext-sRGB.html [hunk 2]
```diff
@@ -298,7 +298,7 @@
   ];
 
   var program = wtu.setupTexturedQuad(gl);
-  gl.uniform1i(gl.getUniformLocation(program, "tex2d"), 0);
+  gl.uniform1i(gl.getUniformLocation(program, "tex"), 0);
```
Comment (zhenyao) [inline sdk/tests/conformance/extensions/ext-sRGB.html]: Good catch.

File: sdk/tests/conformance/extensions/ext-sRGB.html [hunk 3]
```diff
@@ -336,9 +336,9 @@
     // Draw
     var conversions = [
       [   0,   0 ],
-      [  13,  63 ],
-      [  54, 127 ],
-      [ 133, 191 ],
+      [  13,  13 ],
+      [  54,  54 ],
+      [ 133, 133 ],
       [ 255, 255 ]
```
Comment (zhenyao) [inline sdk/tests/conformance/extensions/ext-sRGB.html]: I don't think this is correct.  The target buffer is in sRGB, so after the drawUnitQuad, the buffer's data is in sRGB.  However, ReadPixels is blind to color space, so it reads back the sRGB data.  The original table is correct.

File: sdk/tests/conformance/extensions/ext-sRGB.html [hunk 4]
```diff
@@ -387,15 +387,15 @@
   // Draw
   var conversions = [
     [   0,   0 ],
-    [  13,  63 ],
-    [  54, 127 ],
-    [ 133, 191 ],
+    [  13,  13 ],
+    [  54,  54 ],
+    [ 133, 133 ],
     [ 255, 255 ]
```
Comment (zhenyao) [inline sdk/tests/conformance/extensions/ext-sRGB.html]: Same here, I think the original table is correct.

File: sdk/tests/conformance/extensions/ext-sRGB.html [hunk 5]
```diff
@@ -387,15 +387,15 @@
...
   ];
 
   var program = wtu.setupProgram(gl, ['vertexShader', 'fragmentShader'], ['aPosition'], [0]);
   wtu.setupUnitQuad(gl, 0);
-  var fbo = createsRGBFramebuffer(gl, 4, 4);
+  var fbo = createsRGBFramebuffer(gl, gl.drawingBufferWidth, gl.drawingBufferHeight);
```
Comment (zhenyao) [inline sdk/tests/conformance/extensions/ext-sRGB.html]: Good catch.

File: sdk/tests/conformance/extensions/ext-sRGB.html (hunk 1 above)
Comment (xhcao) [inline sdk/tests/conformance/extensions/ext-sRGB.html]: Thank you, validFormat should be false, 'SRGB_EXT' is definitely a non color-renderable format.

**Comment by kenrussell** [comment]
@xhcao will you be updating this pull request with the needed changes soon?

**Comment by zhenyao** [comment]
lgtm

**Comment by kenrussell** [comment]
LGTM too after the fact.

**Pull request merged**